packages/belos/src/BelosSolverManager.hpp

```diff
@@ -43,6 +43,90 @@
 #ifndef BELOS_SOLVERMANAGER_HPP
 #define BELOS_SOLVERMANAGER_HPP
 
+// These are here for convenience while developing the DII
+// I wanted to keep each cpp file as minimal as possible but not sure where
+// these macros could best live.
+
+// register the managers for a belos lib setup, such as the test Belos_Factory
+#define DEFINE_REGISTER_SOLVER_MANAGER(manager,name)              \
+  namespace Belos {                                               \
+    namespace Details {                                           \
+      void register##manager() {                                  \
+        typedef double ST;                                        \
+        typedef MultiVec<ST> MV;                                  \
+        typedef Operator<ST> OP;                                  \
+        Impl::registerSolverSubclassForTypes<                     \
+          manager<ST,MV,OP>, ST, MV, OP> (name);                  \
+      }                                                           \
+    }                                                             \
+  }
+
+// Currently only doing this for LSQR and RCG complex registration.
+// Note that this macro is only being used for the belos lib.
+// Then we are manually picking the complex managers to be registered.
+// Note that Tpetra below is currently handled differently. We automatically
+// pick up all the classes and then ignore those with no working complex.
+// This difference is not ideal and confusing but I'd like to discuss
+// before developing further.
```

Comment:
Hm, we should think about how to make that more consistent.  Would fixing that help this first pass?

Reply:
Now I've got all the Tpetra and Belos managers registering everything (real and complex) regardless of manager type when complex is on. So I think this makes everything more consistent.